### ISSUE TITLE
q-chi 2dintegration for grazing incidence

### ISSUE THREAD

**Issue by ljr-1959** [silx-kit/pyFAI#2320]
Kudo's for adding grazing support.
Am I correct that there are no chigi_deg and chigi_rad AZMUTHAL_UNITS such that one cannot create a q-chi plot for the fiber/grazing geometry? (Equivalent to transform_polar of pygix)
If so, can this please be added. That transform is powerful when constructing pole-figures...
Thanks

**Comment by EdgarGF93** [comment]
Hello, pyFAI allows now to easily make a qip (in-plane) vs qoop (out-of-plane) representation of the pattern, that is the one that generates the missing wedge, and is sensitive to incident_angle/tilt_angle, etc.
If you want to make a q-chi plot representation, (if I understand you correctly), this is the classical integrate2d result of pyFAI:
```
from pyFAI import load
from pyFAI.calibrant import get_calibrant
from pyFAI import detector_factory
from pyFAI.gui.jupyter import plot2d, subplots
from pyFAI.azimuthalIntegrator import AzimuthalIntegrator

ai = AzimuthalIntegrator(dist=0.1, wavelength=1e-10, detector=detector_factory("Pilatus1M"))
cal = get_calibrant('LaB6')
data = cal.fake_calibration_image(ai=ai)
res2d = ai.integrate2d(data=data, npt_rad=1000)

fig, ax = subplots()
plot2d(res2d, ax=ax)
ax.get_images()[0].set_clim(-1,1)

```
![image](https://github.com/user-attachments/assets/ff9b7a20-c7e7-4881-bde2-1031b8546414)

But this q is the modulus of the scattering vector, so it's the standard reshaping (or caking), to analyze textures in powder diffraction.

If I'm missing something on your request, please tell me :)

**Comment by ljr-1959** [comment]
Thank you for your response.

In grazing incidence, integrate 2D must still reflect the missing wedge, to calculate the proper pole figures with respect to the surface normal, i.e. chi near zero is never measured and thus should not be in the q-chi plot:
The desire is a plot like:
pygix, transform_polar
***@***.***
But integrate2d generates
***@***.***
As I understand integrate2d
res2d = ai.integrate2d(frame2, 300,180, unit=(unit_qtot_A, "chi_deg"), method=("no", "csr", "cython"),mask=ptstmask)

I need to be able to say unit=(unit_qtot_A, "chigi_deg") where the AZUMTHAL_UNIT chigi_deg is based on the arcTan of qoop and qip.

I'm not savvy enough to be able to create the chigi_deg AZUMUTHAL_UNIT with the proper

Peace
-Lee

PS
The above issue also arises in integrate1d if I want to take a sector cut through the, grazing corrected data, not a line cut. The current unit=(unit_qip,unit_qoop) only supports line-cuts. The grazing community almost always ignores this issue and uses 'missing wedge unaware' sector cuts based on the current integrate2d or equivalent, but that doesn't mean we can't do better.
***@***.***

**Comment by EdgarGF93** [comment]
Hello again. We are preparing the release of pyFAI 2025 before the end of January. This could be an interesting new feature for the fiber/grazing incidence module.
Does this look ok to you?

![Image](https://github.com/user-attachments/assets/b457d285-b16a-4a7d-8f30-97d1e1b0b597)

**Comment by ljr-1959** [comment]
Yes, the chi-|q_total| right hand plot is the representation that I could not achieve with the initial integrate2D( unit=(unit_qtot_A, "Chi_deg").

As discussed earlier, there was a similar limitation in integrate1D, when applied to the left-hand plot. Unit=(unit_qip,unit_qoop) did not support sector cuts. Has this been addressed in the 2025 changes?
The desired sector cut is, of course, a line-cut through the right hand image so is available at the cost of computing the entire 2D transform.

Thanks !

-Lee

**Comment by EdgarGF93** [comment]
Yes, with the new units, you can apply sector units.
For example, with a classic grazing incidence representation (qip-qoop):
```
from pyFAI.gui.jupyter import plot1d, plot2d, subplots
import matplotlib.pyplot as plt
from pyFAI.test.utilstest import UtilsTest
from pyFAI import load
import fabio

if __name__ == "__main__":
    fi = load(UtilsTest.getimage("LaB6_3.poni"), type_="pyFAI.integrator.fiber.FiberIntegrator")
    data = fabio.open(UtilsTest.getimage("Y6.edf")).data
    air = fabio.open(UtilsTest.getimage("air.edf")).data
    data_clean = data - 0.03 * air
    sample_orientation = 6
        
    res2d_0 = fi.integrate2d_grazing_incidence(data=data_clean, sample_orientation=sample_orientation,
                                            )
    res1d_0 = fi.integrate1d_grazing_incidence(data=data_clean, sample_orientation=sample_orientation,
                                            ip_range=[-1,1], oop_range=[0,20],
                                            npt_ip=100, npt_oop=500,
                                            vertical_integration=True,
                                            )
    res1d_1 = fi.integrate1d_grazing_incidence(data=data_clean, sample_orientation=sample_orientation,
                                            ip_range=[-17.5,17.5], oop_range=[5,6.5],
                                            npt_ip=500, npt_oop=100,
                                            vertical_integration=False,
                                            ) 
    
    fig, ax = subplots(ncols=3)
    plot2d(result=res2d_0, ax=ax[0])
    img = ax[0].get_images()[0]
    img.set_cmap("viridis")
    img.set_clim(20,200)
    plot1d(result=res1d_0, ax=ax[1])
    plot1d(result=res1d_1, ax=ax[2])
    
    plt.show()
```

![Image](https://github.com/user-attachments/assets/15a3f5cb-1f98-4b25-b612-177ed216b423)

**Comment by EdgarGF93** [comment]
to be fixed in #2380

**Comment by EdgarGF93** [comment]
Closed with 2025.01 release
